build.sbt

```diff
@@ -102,7 +103,55 @@ inThisBuild(
         }
         .mkString(" ")
     },
-    resolvers += Resolver.bintrayRepo("scalacenter", "releases")
+    resolvers += Resolver.bintrayRepo("scalacenter", "releases"),
+    githubActionsWorkflow := {
+      Obj(
+        "name" -> Str("CI"),
+        "on" -> Arr(Str("push"), Str("pull_request")),
+        "jobs" -> Obj(
+          "Scalafmt" -> githubActionsJob(
+            "Formatting",
+            "./bin/scalafmt --test"
+          ),
+          "Scalafix" -> githubActionsJob(
+            "Linting",
+            "csbt scalafixCheck githubActionsCheck"
+          ),
+          "Docusaurus" -> githubActionsJob(
+            "Website",
+            "csbt docs/docusaurusCreateSite"
+          ),
+          "Metals" -> githubActionsJob(
+            "Run tests",
+            "csbt 'unit/testOnly -- tests.${{ matrix.test }}'",
+            githubActionsMatrix(
+              "test",
+              githubActionsPartition(
+                baseDirectory.in(ThisBuild).value /
+                  "tests" / "unit" / "src" / "test" / "scala" / "tests",
+                partitionCount = Some(3)
+              )
+            )
+          ),
+          "Mtags" -> githubActionsJob(
+            "Run tests",
+            "csbt ++${{  matrix.scala }} cross/test",
+            githubActionsMatrix("scala", V.supportedScalaVersions)
```

Comment:
@milessabin One option for syncing the CI with `crossScalaVersions` is to automatically generate `.travis.yml` from build.sbt. In this PR we're using GitHub Actions but the same approach should work for Travis CI.

Reply:
@olafurpg you must be pulling my leg ... @djspiewak told me this is "fundamentally impossible"! ;-)

But seriously ... thanks for the pointer :-)

Reply:
It doesn't even seem necessary to print YAML, you can print JSON as long as the filename stays `travis.yml`

Reply:
> In this PR we're using GitHub Actions but the same approach should work for Travis CI.

It doesn't work with travisci unless you commit the results. If Github Actions has multi-staged builds without a need to hard-commit results, then it would be capable of this, but Travis' architecture doesn't support that.

Reply:
> It doesn't work with travisci unless you commit the results.

When did I ever say I didn't want to commit the results!?!